docs/samples/Microsoft.ML.Samples/Dynamic/LightGBM/LightGBMRegression.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic.LightGBM
+{
+    class LightGbmRegression
+    {
+        public static void LightGbmRegressionExample()
+        {
+            // Downloading a regression dataset from github.com/dotnet/machinelearning
+            // this will create a housing.txt file in the filsystem this code will run
+            // you can open the file to see the data. 
+            string dataFile = SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset();
+
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Creating a data reader, based on the format of the data
+            // The data is tab separated with all numeric columns.
+            // The first column being the label and rest are numeric features
+            // Here only seven numeric columns are used as features
+            var dataView = mlContext.Data.ReadFromTextFile(dataFile, new TextLoader.Arguments
+            {
+                Separators = new[] { '\t' },
+                HasHeader = true,
+                Columns = new[]
+               {
+                    new TextLoader.Column("Label", DataKind.R4, 0),
+                    new TextLoader.Column("Features", DataKind.R4, 1, 6)
+                }
+            });
+
+            //////////////////// Data Preview ////////////////////
+            // MedianHomeValue    CrimesPerCapita    PercentResidental    PercentNonRetail    CharlesRiver    NitricOxides    RoomsPerDwelling    PercentPre40s
+            // 24.00              0.00632            18.00                2.310               0               0.5380          6.5750              65.20
+            // 21.60              0.02731            00.00                7.070               0               0.4690          6.4210              78.90
+            // 34.70              0.02729            00.00                7.070               0               0.4690          7.1850              61.10
+
+            var (trainData, testData) = mlContext.Regression.TrainTestSplit(dataView, testFraction: 0.1);
+
+            // Create the estimator, here we only need LightGbm trainer 
+            // as data is already processed in a form consumable by the trainer
+            var pipeline = mlContext.Regression.Trainers.LightGbm(
+                                            numLeaves: 4,
+                                            minDataPerLeaf: 6,
+                                            learningRate: 0.001);
+
+            // Fit this pipeline to the training data
+            var model = pipeline.Fit(trainData);
+
+            // Check the weights that the model learned
+            VBuffer<float> weights = default;
+            model.Model.GetFeatureWeights(ref weights);
```

Comment:
>model.Model.GetFeatureWeights(ref weights); [](start = 12, length = 43)

That thing doesn't have much sense for me. We train tree, right? What this weights mean? How about 
`model.Model.TrainedTreeEnsemble.Trees[0].LeafValues` and show values in it.

Reply:
Is it not good to show the feature importance based on the information gain used during training?

---
In reply to: [255286511](https://github.com/dotnet/machinelearning/pull/2476#discussion_r255286511) [](ancestors = 255286511)